src/TSTransformer/util/arrayBindingPatternContainsHoists.ts

```diff
@@ -0,0 +1,21 @@
+import { TransformState } from "TSTransformer/classes/TransformState";
+import { checkVariableHoist } from "TSTransformer/util/checkVariableHoist";
+import ts from "typescript";
+
+export function arrayBindingPatternContainsHoists(
+	state: TransformState,
+	arrayBindingPattern: ts.ArrayBindingPattern,
+): boolean {
+	for (const element of arrayBindingPattern.elements) {
+		if (ts.isBindingElement(element) && ts.isIdentifier(element.name)) {
+			const symbol = state.typeChecker.getSymbolAtLocation(element.name);
+			if (symbol) {
+				checkVariableHoist(state, element.name, symbol);
+				if (state.isHoisted.get(symbol)) {
+					return true;
+				}
```

Comment:
Could do with explainer comment to indicate the hoist is _marked_ in checkVariableHost first, then checked from the state. Or maybe I misunderstood, and that would also illustrate a comment is necessary to explain it.